src/gradata/_cloud_sync.py

```
@@ -0,0 +1,222 @@
+    req = urllib.request.Request(
+        url,
+        data=body,
+        method="POST",
+        headers={
+            "apikey": key,
+            "Authorization": f"Bearer {key}",
+            "Content-Type": "application/json",
+            # Upsert on primary key so repeat pushes are idempotent.
+            "Prefer": "resolution=merge-duplicates,return=minimal",
+        },
+    )
```

Ruff (0.15.10)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)